### DISCUSSION TITLE
Minifloat

### DISCUSSION THREAD

**Discussion by juancolonna** [logisim-evolution/logisim-evolution#1608] [Ideas]
Dear developers, first I'd like to thank you for the amazing work you are doing developing Logisim. Second, I would like to request a new feature if possible.  

I use Logisim as a teaching tool with my students.  I usually explain initial concepts using 8-bit logic, in a simple way.  I often using minifloat (8-bit float) to explain floating number concepts (scale, overflow, etc.).  Floating arithmetic blocks don't have the 8-bit option, eg: addition, multiplication, subtraction, etc., and for me it would be very useful.  Would it be possible to implement 8-bit float operations?  I would be very grateful.

At,

**Comment by maehne** [comment]
Thanks! We are always glad if our users like our tool and find it useful as a teaching tool!

Your suggestion is nice and one may also consider to support 16 bit floats at the same time. Implementation of the floating point arithmetic blocks is currently based on Java's `double` data type. Smaller floats would need to get converted manually back and forth to this representation (e.g., see [here](https://stackoverflow.com/questions/37105764/how-to-convert-to-from-an-8-bit-float-representation)). One would not only need to adapt these blocks, but also the port and probe components (so that the signal values can be correctly shown when choosing the radix "Float".

PRs in this direction are welcome! As a starting point, have a look into the `Fp*.java` files in `src/main/java/com/cburch/logisim/std/arith/`.
